package ohtu;

import java.util.Random;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;

public class Tester {
    public static void main(String[] args) {
        WebDriver driver = new ChromeDriver();

        // Huom! samaan aikaan toisesta terminaalista ./gradlew run jotta toimii :)
        
        // LOGIN VALIDATION TEST WITH CORRECT USERNAME - INCORRECT PASSWORD
        driver.get("http://localhost:4567");

        sleep(1);

        WebElement element = driver.findElement(By.linkText("login"));
        element.click();

        sleep(2);

        element = driver.findElement(By.name("username"));
        element.sendKeys("pekka");
        element = driver.findElement(By.name("password"));
        element.sendKeys("wrong");

        sleep(2);

        element.submit();

        sleep(3);
        // LOGIN VALIDATION TEST WITH CORRECT USERNAME - INCORRECT PASSWORD

        // REGISTER & LOGOUT TEST
        driver.get("http://localhost:4567");
        
        sleep(2);
        
        element = driver.findElement(By.linkText("register new user"));
        element.click();

        sleep(2);

        Random r = new Random();
        element = driver.findElement(By.name("username"));
        element.sendKeys("pekka" + r.nextInt(100000));
        element = driver.findElement(By.name("password"));
        element.sendKeys("akkep");
        element = driver.findElement(By.name("passwordConfirmation"));
        element.sendKeys("akkep");
        
        sleep(2);

        element.submit();

        sleep(2);

        element = driver.findElement(By.linkText("continue to application mainpage"));
        element.click();

        sleep(2);

        element = driver.findElement(By.linkText("logout"));
        element.click();

        sleep(2);

        // REGISTER & LOGOUT TEST
        
        driver.quit();
    }
    
    private static void sleep(int n){
        try{
            Thread.sleep(n*1000);
        } catch(Exception e){}
    }
}
